src/components/HourChart/HourChart.jsx

```diff
@@ -364,7 +371,13 @@ export default class HourChart extends PureComponent {
    * @return {React.Component} The rendered container
    */
   render() {
-    const { width, id, height } = this.props;
+    const { dataByHour } = this.prepareData(this.props);
```

Comment:
This is slow since it gets called every single render. It is better to use this.visComponents and handle if this.visComponents hasn't been made yet. Otherwise we will experience sluggish hover behavior as it does extra computation in the render method.